conduit/pipeline/pipeline.go

```diff
@@ -532,6 +539,13 @@ func (p *pipelineImpl) Wait() {
 	p.wg.Wait()
 }
 
+func (p *pipelineImpl) Status() (Status, error) {
+	rnd := atomic.LoadUint64(&p.pipelineMetadata.NextRound)
```

Comment:
Docs seem to suggest using `Load()` here: https://pkg.go.dev/sync/atomic#LoadUint64

Also do we need to atomically increment it here? https://github.com/winder/conduit/blob/3c92877960bbc00f26070a3076a653e3c5da46f7/conduit/pipeline/pipeline.go#L462

Reply:
Zeph is working on that code, so I didn't want to touch it here. We're careful to avoid using `pipelineMetadata` in other go routines, so I don't think we need any special handling there.

I saw that comment in the docs, but wasn't sure I wanted to change types for something so simple. Looking again [and now finding the bugs reference](https://pkg.go.dev/sync/atomic#pkg-note-BUG), maybe that would be a better option.

Reply:
@tzaffi what do you think about this?

Reply:
I think you should be able to use `p.pipeline.NextRound.Load()` here without interfering so much with #128. However, I do agree that refactoring all the usages is worthwhile and I created an issue: #140 .

Reply:
I don't feel very strongly since getting `round` wrong is probably not critical to most applications. I'll defer judgement to you.

Reply:
@algochoi @tzaffi The comments about endienness and caveats about what processors support 64 bit atomic operations seemed dangerous, so I went ahead and switched to a simple mutex.

A channel could be used to keep the pipeline loop lock-free, but given the lock only needs to be held for the duration of a copy or an assignment, it didn't seem like that would have any real impact to performance.